src/WebJobs.Script/Rpc/LanguageWorkerChannelManager.cs

```diff
@@ -164,50 +175,57 @@ internal void ScheduleShutdownStandbyChannels()
             _workerRuntime = _workerRuntime ?? _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
             if (!string.IsNullOrEmpty(_workerRuntime))
             {
-                var standbyChannels = _workerChannels.Where(ch => ch.Key.ToLower() != _workerRuntime.ToLower()).ToList();
-                for (int i = 0; i < standbyChannels.Count(); i++)
+                var standbyWorkerChannels = _workerChannels.Where(ch => ch.Key.ToLower() != _workerRuntime.ToLower());
+                foreach (var runtime in standbyWorkerChannels)
                 {
-                    _logger.LogInformation("Disposing standby channel for runtime:{language}", standbyChannels.ElementAt(i).Key);
-                    standbyChannels.ElementAt(i).Value.Dispose();
-                    _workerChannels.Remove(standbyChannels.ElementAt(i).Key);
+                    _logger.LogInformation("Disposing standby channel for runtime:{language}", runtime.Key);
+                    foreach (var channel in runtime.Value.ToList())
+                    {
+                        if (channel != null)
+                        {
+                            channel.Dispose();
+                            runtime.Value.Remove(channel);
+                        }
+                    }
                 }
             }
         }
 
-        public void ShutdownStandbyChannels(IEnumerable<FunctionMetadata> functions)
+        public void ShutdownChannels()
         {
-            if (_environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode) == "1")
+            foreach (string runtime in _workerChannels.Keys)
             {
-                return;
-            }
-            _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName) ?? Utility.GetWorkerRuntime(functions);
-            _logger.LogInformation("WorkerRuntime: {workerRuntime}. Will shutdown other standby channels", _workerRuntime);
-            if (string.IsNullOrEmpty(_workerRuntime))
-            {
-                ShutdownChannels();
-                return;
-            }
-            else
-            {
-                ScheduleShutdownStandbyChannels();
+                _logger.LogInformation("Shutting down language worker channels for runtime:{runtime}", runtime);
+                foreach (var channel in _workerChannels[runtime].ToList())
+                {
+                    if (channel != null)
+                    {
+                        channel.Dispose();
+                        _workerChannels[runtime].Remove(channel);
```

Comment:
Should the channel be removed prior to disposal here?

Reply:
channel needs to be diposed first to make sure all the subscriptions and the process attached is disposed before removing the reference to the channel